components/org.wso2.carbon.identity.sts.passive/src/main/java/org/wso2/carbon/identity/sts/passive/custom/provider/CustomCryptoProvider.java

```diff
@@ -70,31 +72,16 @@ public CustomCryptoProvider(Properties properties, ClassLoader loader,
     protected KeyStore load(InputStream input, String storepass, String provider, String type)
             throws WSSecurityException {
 
-        KeyStore keyStore;
-
-        String tenantId = this.properties.getProperty(TENANT_ID_PROP);
-        String keyStoreName = this.properties.getProperty(KEY_STORE_NAME_PROP);
-
         log.debug("Loading keystore...");
-        if (!String.valueOf(MultitenantConstants.SUPER_TENANT_ID).equals(tenantId)
-                && keyStoreName != null) {
-            // Loads the keystore in a custom way since the tenant keystore does not have a location.
-            if (log.isDebugEnabled()) {
-                log.debug("Loading keystore for tenant with id: " + tenantId + ".");
-            }
-            KeyStoreManager keyStoreManager = KeyStoreManager.getInstance(Integer.parseInt(tenantId));
-            try {
-                keyStore = keyStoreManager.getKeyStore(keyStoreName);
-            } catch (Exception exception) {
-                throw new WSSecurityException(WSSecurityException.ErrorCode.FAILURE, exception, "failedCredentialLoad");
-            }
-        } else {
-            // Loads the keystore in the default way since the keystore has a location.
+        try {
+            String tenantDomain = PrivilegedCarbonContext.getThreadLocalCarbonContext().getTenantDomain();
+            return IdentityKeyStoreResolver.getInstance().getKeyStore(
+                    tenantDomain, IdentityKeyStoreResolverConstants.InboundProtocol.WS_FEDERATION);
+        } catch (IdentityKeyStoreResolverException e) {
             if (log.isDebugEnabled()) {
```

Comment:
No need to check for `log.isDebugEnabled()` since there's no string operations in  log.